bin/create-action.js

```diff
@@ -0,0 +1,169 @@
+// @ts-check
+
+const fs = require('fs')
+const path = require('path')
+const { promisify } = require('util')
+const minimist = require('minimist')
+const { prompt } = require('enquirer')
+const colors = require('./colors.json')
+const icons = require('./feather-icons.json')
+
+const readFile = promisify(fs.readFile)
+const writeFile = promisify(fs.writeFile)
+const mkdir = promisify(fs.mkdir)
+
+/**
+ * Reads a template file from disk.
+ *
+ * @param {string} filename The template filename to read.
+ * @returns {Promise<string>} The template file string contents.
+ */
+const readTemplate = filename => {
+  const templateDir = path.join(__dirname, 'template')
+  return readFile(path.join(templateDir, filename), 'utf8')
+}
+
+/**
+ * A predicate function to ensure a string is not empty.
+ *
+ * @param {string} value The string value.
+ * @returns {boolean} Whether the string is empty or not.
+ */
+const isNotEmpty = value => value.length > 0
+
+/**
+ * The options object returned from the CLI questionnaire prompt.
+ * @typedef {object} PromptAnswers
+ * @property {string} name The action name.
+ * @property {string} description The action description.
+ * @property {string} icon The feather icon name. See `bin/feather-icons.json` for options.
+ * @property {string} color The GitHub Action color. See `bin/colors.json` for options.
+ */
+
+/**
+ * Prompts the user with a questionnaire to get key metadata for the GitHub Action.
+ *
+ * @returns {Promise<PromptAnswers>} An object containing prompt answers.
+ */
+const getActionMetadata = async () => {
+  try {
+    return await prompt([
+      {
+        type: 'input',
+        name: 'name',
+        message: 'What is the name of your action?',
+        initial: 'Your action name',
+        validate: isNotEmpty
+      },
+      {
+        type: 'input',
+        name: 'description',
+        message: 'What is a short description of your action?',
+        initial: 'A description of your action',
+        validate: isNotEmpty
+      },
+      {
+        type: 'autocomplete',
+        name: 'icon',
+        message: 'Choose an icon for your action. Visit https://feathericons.com for a visual reference.',
+        choices: icons
+      },
+      {
+        type: 'autocomplete',
+        name: 'color',
+        message: 'Choose a background color background color used in the visual workflow editor for your action.',
+        choices: colors
+      }
+    ])
+  } catch (err) {
+    // When prompt() rejects, that means the user has pressed ctrl+c to cancel input.
+    throw new Error('Cancelled. Maybe next time!')
```

Comment:
Do we need to show a message to the user if they've canceled the CLI prompt? Could just make this empty.

Reply:
Nah, I don't think so. I think we're better off making it empty for a cleaner exitt.